serviceerror/convert.go

```diff
@@ -42,6 +42,14 @@ func ToStatus(err error) *status.Status {
 	if svcerr, ok := err.(ServiceError); ok {
 		return svcerr.Status()
 	}
+	// err does not implement ServiceError directly, but check if it wraps it.
+	// This path does more copying so prefer to return a ServiceError directly if possible.
+	var svcerr ServiceError
+	if errors.As(err, &svcerr) {
+		s := svcerr.Status().Proto()
```

Comment:
Just double-checking, this does a copy right? In any case, can we verify in the test that the original message is unmodified?

Reply:
I'm not sure I would say it does a "copy". It asks the ServiceError to serialize itself into a google.golang.org/genproto/googleapis/rpc/status.Status (proto message). The it replaces one field in the proto message and constructs a google.golang.org/grpc/status.Status from that. Those both involve copying some fields, but we're moving between different types. Specifically, there's only ever one ServiceError.

Reply:
I think the underlying implementation of Proto() does a Clone() so original msg should be unmodified.